service/history/replication/raw_task_converter.go

```diff
@@ -655,13 +655,30 @@ func newSyncVersionedTransitionTaskConverter(
 		targetClusterID:  targetClusterID,
 		executionManager: executionManager,
 		mutableState:     mutableState,
-		logger:           logger,
+		logger: log.NewLazyLogger(
+			logger,
+			func() []tag.Tag {
+				return []tag.Tag{
+					tag.WorkflowNamespaceID(taskInfo.NamespaceID),
+					tag.WorkflowID(taskInfo.WorkflowID),
+					tag.WorkflowRunID(taskInfo.RunID),
+					tag.TaskKey(taskInfo.GetKey()),
+					tag.Task(taskInfo),
+				}
+			},
+		),
 	}
 }
 
 func (c *syncVersionedTransitionTaskConverter) convert() (*replicationspb.ReplicationTask, error) {
-	progress := c.replicationCache.Get(c.taskInfo.RunID, c.targetClusterID)
 
+	// If workflow is not on any versionedTransition (in an unknown state from state-based replication perspective),
+	// we can't convert this raw task to a replication task, instead we need to rely on its task equivalents.
+	if len(c.mutableState.GetExecutionInfo().TransitionHistory) == 0 {
```

Comment:
When we disable state-based replication, will we clear `TransitionHistory` of all existing workflows?

Reply:
yeah I will do that in a later PR to clear all fields related to VersionedTransition is an event-based replication task is applied to a workflow.